test/image/mocks/gl2d_parcoords_constraints.json

```diff
@@ -0,0 +1,178 @@
+{
+  "layout": {
+    "width": 1000,
+    "height": 400
+  },
+
+  "data": [{
+    "type": "parcoords",
+
+    "line": {
+      "showscale": true,
+      "cmin": 0,
+      "cmax": 20,
+      "color": [
+        0, 1, 2, 3,
+        4, 5, 6, 7, 8, 9, 0, 1,
+        0, 0, 0, 0,
+        2, 3, 4, 5,
+        6, 7, 8, 9,
+        0, 1, 2, 3,
+        4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9,
+        0, 1, 2, 3,
+        4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9
+      ]
+    },
+
+    "dimensions": [
+      {
+        "label": "±1.1",
+        "constraintrange": [-1.1, 1.1],
+        "range": [-2, 10],
+        "values": [
+          -1.1, 1.1, -1.10001, 1.10001,
+          -0.7, -0.7, -0.7, -0.7, 0.7, 0.7, 0.7, 0.7,
+          0, 0, 0, 0,
+          0, 0, 0, 0,
+          0, 0, 0, 0,
+          0, 0, 0, 0,
+          0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+          0, 0, 0, 0,
+          0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+        ]
+      },
+      {
+        "label": "±[5,10],0",
+        "constraintrange": [[-10, -5], [-1, 1], [5, 10]],
+        "values": [
+          -9, -9, -7, -7,
+          -10.001, 10.001, -4.98, 4.98, -10, 10, -5, 5,
```

Comment:
Nearly every sample on this plot is testing something... the values arrays are broken into lines based on which dimension is being tested. This dimension is testing that precision on the outer bounds of the constraint range is limited by floating point (OK, I could have added another zero or two, `10.00001` or so...) and precision in the gaps between selection ranges is not as good, but this test (`4.98`) shows at least one thousandth of the range. The number of vertical pixels is 2048, and as expected if I move to `4.99` one of these values gets incorrectly marked valid. Notice that for this test to pass, all the lines going to `-0.7` in the preceding and following dimensions should be grey, and all the lines going to `0.7` should be colored.

Reply:
I didn't notice any performance issue with making the filter mask a fixed 2048 height, since it only has a width of 8. In principle we should be able to double that if more precision is required.